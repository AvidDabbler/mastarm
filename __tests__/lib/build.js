/* globals afterEach, beforeEach, describe, it, expect, jasmine */

const build = require('../../lib/build')
const util = require('../test-utils/util.js')

const originalTimeout = jasmine.DEFAULT_TIMEOUT_INTERVAL

describe('build', () => {
  const mockDir = util.mockDir

  beforeEach(() => {
    jasmine.DEFAULT_TIMEOUT_INTERVAL = 20000
  })
  afterEach(() => {
    jasmine.DEFAULT_TIMEOUT_INTERVAL = originalTimeout
  })

  describe('development', () => {
    it('should transform js', async () => {
      const [result] = await build({
        config: {},
        files: [[`${mockDir}/index.js`]]
      })

      console.log(result)

      expect(result.toString().indexOf('MockTestComponentUniqueName')).not.toBe(-1)
    })

    it('should transform css', async () => {
      const [result] = await build({
        config: {},
        files: [[`${mockDir}/index.css`]]
      })

      const css = result.css
      expect(css.indexOf('criticalClass')).toBeGreaterThan(-1)
    })
  })

  describe('production', () => {
    it('should transform and minify js', async () => {
      const output = await build({
        config: {},
        env: 'production',
        files: [
          [`${mockDir}/index.js`],
          [`${mockDir}/index.css`]
        ],
        minify: true
      })

      expect(output[0].toString().indexOf('MockTestComponentUniqueName')).not.toBe(-1)
      expect(output[1].css.indexOf('criticalClass')).not.toBe(-1)
    })
  })
})
