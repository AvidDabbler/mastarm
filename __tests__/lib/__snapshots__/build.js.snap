// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`build development should transform css 1`] = `86`;

exports[`build development should transform js 1`] = `157929`;

exports[`build production should transform and minify js and css 1`] = `156127`;

exports[`build production should transform and minify js and css 2`] = `86`;
